packages/devtools_app/lib/src/ui/search.dart

```diff
@@ -618,26 +616,26 @@ typedef HighlightAutoComplete = Function(
 /// Callback for clearing the search field.
 typedef ClearSearchField = Function(
   SearchControllerMixin controller, {
-  bool force,
+  bool? force,
 });
 
 /// Provided by clients to specify where the autocomplete overlay should be
 /// positioned relative to the input text.
 typedef OverlayXPositionBuilder = double Function(
   String inputValue,
-  TextStyle inputStyle,
+  TextStyle? inputStyle,
 );
 
 // TODO(elliette) Consider refactoring this mixin to be a widget. See discussion
 // at https://github.com/flutter/devtools/pull/3532#discussion_r767015567.
 mixin SearchFieldMixin<T extends StatefulWidget>
     on AutoDisposeMixin<T>, State<T> {
-  TextEditingController searchTextFieldController;
-  FocusNode _searchFieldFocusNode;
-  FocusNode _rawKeyboardFocusNode;
-  SelectAutoComplete _onSelection;
+  TextEditingController? searchTextFieldController;
+  FocusNode? _searchFieldFocusNode;
+  FocusNode? _rawKeyboardFocusNode;
```

Comment:
these can all be late. Anything that is initialized in initState can be late

Reply:
done